src/NServiceBus.NHibernate/SynchronizedStorage/INHibernateSynchronizedStorageSession.cs

```diff
@@ -1,9 +1,13 @@
 ﻿namespace NServiceBus
 {
+    using System;
+    using System.Threading.Tasks;
     using global::NHibernate;
 
     interface INHibernateSynchronizedStorageSession
     {
         ISession Session { get; }
+
+        void RegisterCommitHook(Func<Task> callback);
```

Comment:
Would there be value in trying to align with NH event listener lingo

http://nhibernate.info/doc/nhibernate-reference/events.html

In that case perhaps: `OnCommit(...)` ?